src/state_management.cpp

```diff
@@ -18,12 +18,19 @@ namespace ara { namespace sm {
         } else {
             //don't reset??
         }
+
+        if (stateClient != nullptr) {
+            internalState = stateClient->requestedSMState;
+        }
     }
 
     void StateManagement::Kill() {
         killFlag = true;
     }
 
-    StateManagement::StateManagement() :
-        myUpdateRequest{com::UpdateRequest()}, myNetworkHandle{com::NetworkHandle()}, killFlag{false} {}
+    StateManagement::StateManagement(exec::StateClient * sc) :
```

Comment:
shall be 'exec::StateClient* stateClient' or 'exec::StateClient *stateClient'. Nowadays, I think the first option is common

Reply:
Done